File: C#/Solutions/Problem016/Log.cs
using System.Collections.Generic;

namespace DailyCodingProblems.Solutions.Problem016
{
    /// <summary>
    /// Log that contains the orderIds
    /// </summary>
    public class Log
    {
        private readonly IList<int> _log;

        public int Size { get; }

        public Log(int size)
        {
            _log = new List<int>();
            Size = size;
        }

        /// <summary>
        /// Adds the order_id to the log
        /// </summary>
        /// <param name="orderId"></param>
        public void Record(int orderId)
        {
            if (_log.Count == Size)
                _log.RemoveAt(0);

            _log.Add(orderId);
        }

        /// <summary>
        /// Gets the ith last element from the log
        /// </summary>
        /// <param name="i"></param>
        /// <returns></returns>
        public int GetLast(int i)
        {
            return _log[^i];
        }
    }
}

File: Java/Solutions/Problem016/Log.java
package Problem016;

import java.util.ArrayList;
import java.util.List;

/**
 * Log that contains the orderIds
 */
public class Log {
    private final List<Integer> log;
    private int size;

    public Log(int size) {
        log = new ArrayList<>();
        this.size = size;
    }

    public void record(int orderId) {
        if (log.size() == size)
            log.remove(0);

        log.add(orderId);
    }

    public int getLast(int i) {
        return log.get(log.size() - i);
    }
}
